### DISCUSSION TITLE
Integrating with angular application

### DISCUSSION THREAD

**Discussion by im-saro** [stoplightio/spectral#1578] [Q&A]
is it possible to integrate spectral with angular application?. do we have any document on how to achieve this.

**Comment by ghost** [comment]
did you find any clue to run this as a service OR integrate into existing application?

**Comment by magicmatatjahu** [comment]
What exactly do you have a problem with? You are able to use Spectral in any frontend application. The only problem you may have is with some of the dependencies that Spectral uses, such as native NodeJS modules or packages that can only be used on the NodeJS side, not the browser side - but they can always be filtered through the webpack (or any other bundler) so it doesn't add them to the final bundle.

I'm currently integrating it with a React application on this PR (https://github.com/asyncapi/studio/pull/434) and see no problem so far.

**Comment by ghost** [comment]
I am actually trying to find a way to run this as a service. Let say if a user send open API specification file as http request then this service apply a ruleset/lint and give back the results. Any idea if spectral already do the same over http service?

**Reply by magicmatatjahu** [reply]
Spectral underneath (as far as I know, but I could be wrong) does not "support" specification posted somewhere on the Internet. You have to download these specification yourself on the browser side, then pass the content of a given http(s) to Spectral and specify its source ("path"), something like:

```js
const spectral = new Spectral(...)
spectral.setRuleset(...);

// source is a http link to the specification 
const content = (await fetch(source, ...options)).text();

// third argument is important, by "source" Spectral Ref Parser will dereference local refs in linked specification
const document = new Document(content, Yaml, source);
const results = await spectral.run(document);
```

If you wanna run Spectral in server, above logic is this same. At least I do it that way and it works 😄

**Reply by magicmatatjahu** [reply]
If user send the specification as JSON or string to the server/browser textarea then the logic is identical, but you don't make fetching.
